numpy/core/src/multiarray/item_selection.c

```diff
@@ -2128,6 +2129,68 @@ count_nonzero_bytes_384(const npy_uint64 * w)
     return r;
 }
 
+#if NPY_SIMD
+
+/* Count the zero bytes between `*d` and `end`, updating `*d` to point to where to keep counting from. */
+static NPY_INLINE NPY_GCC_OPT_3 npyv_u8
+count_zero_bytes_u8(const npy_uint8 **d, const npy_uint8 *end)
+{
+    const npyv_u8 vone = npyv_setall_u8(1);
+    const npyv_u8 vzero = npyv_zero_u8();
+
+    npy_intp n = 0;
+    npyv_u8 vsum8 = npyv_zero_u8();
+    while (*d < end && n <= 0xFE) {
+        npyv_u8 vt = npyv_cvt_u8_b8(npyv_cmpeq_u8(npyv_load_u8(*d), vzero));
+        vt = npyv_and_u8(vt, vone);
+        vsum8 = npyv_add_u8(vsum8, vt);
+        *d += npyv_nlanes_u8;
+        n += npyv_nlanes_u8;
```

Comment:
I hadn't noticed that you changed this from `1` to `nlanes`. I'd argue this change makes it less performant, and that strictly only the last accumulation into the 2**32-bit counters needs this adjustment to avoid overflow.

Reply:
My goal was for `n` in all these functions to mean "the largest possible value in a single lane. Perhaps `lane_max` would be a better name.

Reply:
what a good name, in the `count_zero_bytes_u16` and `count_zero_bytes_u32` lane_max needs to multiplied by 2 because the expand and add operation. The final algorithm may looks like:
![image](https://user-images.githubusercontent.com/15245051/102326974-0845e080-3fc0-11eb-8d99-1f90d67dbae3.png)

Reply:
Yeah, I was just thinking about the factor of two, that makes sense.

It might be a good idea to add that drawing as ascii art so that it doesn't get lost.

Reply:
Do you mean something like this?
```
/*
The one loop process is as follows(take SSE2 with 128bits vector for example):
         |------------16 lanes---------|          
[vsum8]  255 255 255 ... 255 255 255 255 count_zero_bytes_u8: counting 255*16 elements
                         !!
          |------------8 lanes---------|          
[vsum16]  65535 65535 65535 ...   65535  count_zero_bytes_u16: counting 65535*8 elements
                         !!
          |------------4 lanes---------|          
[vsum32]  1073741824   ...   1073741824  count_zero_bytes_u32(overflow control): counting 2**32-1 elements
                         !!
                       2**32-1           count_zero_bytes
*/
```